gpflow/base.py

```diff
@@ -52,7 +56,7 @@ class PriorOn(Enum):
 class Parameter(tf.Module):
     def __init__(
         self,
-        value,
+        value: Union[_ArrayOrScalar, _TensorTypeExternal, "Parameter"],
```

Comment:
is it worth having an alias for this union. Its used a lot

Reply:
probs

Reply:
though I'm running out of tensor-related names

Reply:
How is it different from `TensorLike` (or `TensorType`, whichever one it is now)?

Reply:
it's currently `Union[_Array, TensorType]`

Reply:
yeah but I can do that anywhere else that expects a tensor-like thing, too.

Reply:
you can't do `[1, 2, 3] / 2` but you can do `tf.constant([1, 2, 3]) / 2`

Reply:
urgh. that's the part I really dislike about adding types, I think this is getting to the point where it makes the code less clear and less readable, just to satisfy mypy under all possible conditions...

Reply:
i anticipate this is one of the harder, if not the hardest part of adding types to gpflow - sorting out the tensors, but I hope that once we've done it once, we won't need to do it again. I've just changed the type aliases a little, and aside from the `TensorLike` which is needed for multipledispatch, there's only two standard tensor types: `Union[tf.Tensor, tf.Variable, np.ndarray, "Parameter"]` and that plus `Union[int, float, Sequence[Any]]` (though in some cases we use `tf.Tensor`, `Parameter` or `tf.Variable` explicitly)

Reply:
I hope I can demonstrate to you this is doable in a neat way